subsys/emds/emds_flash_new.h

```diff
@@ -83,17 +115,38 @@ int emds_flash_init(struct emds_partition *partition);
  * @brief Scan the emergency data storage partition for valid snapshots.
  *
  * This function scans the specified partition for valid snapshots and populates
- * the provided metadata structure with the found snapshot metadata with the biggest
+ * the candidate structure with the found snapshot metadata with the biggest
  * fresh_cnt value and valid metadata and snapshot crc values.
  *
  * @param partition Pointer to the emergency data storage partition structure.
- * @param metadata Pointer to the metadata structure to be filled with found snapshot data.
+ * @param candidate Pointer to the emergency data storage snapshot candidate structure
  *
  * @retval 0 on success.
  * @retval -EINVAL if the partition is not initialized or if an error occurs during reading.
  */
 int emds_flash_scan_partition(const struct emds_partition *partition,
-				  struct emds_snapshot_metadata *metadata);
+			      struct emds_snapshot_candidate *candidate);
+
+/** * @brief Allocate a new snapshot in the emergency data storage partition.
+ *
+ * This function allocates a new snapshot in the specified partition based on the
+ * freshest snapshot found in the partition.
+ *
+ * @param partition Pointer to the emergency data storage partition structure.
+ * @param freshest_snapshot Pointer to the freshest snapshot candidate structure.
+ * @param allocated_snapshot Pointer to the emergency data storage snapshot candidate structure
+ * that will be filled with the allocated snapshot metadata.
+ * @param data_size The size of the data to be allocated.
```

Comment:
Considering metadata size is substracted from here, can you add this?
```suggestion
 * @param data_size The size of the data to be allocated including the size of the metadata.
```